src/components/calculator.js

```diff
@@ -1,42 +1,59 @@
+/* eslint-disable react/prefer-stateless-function, jsx-a11y/click-events-have-key-events, react/no-unused-state, max-len, react/jsx-one-expression-per-line, jsx-a11y/control-has-associated-label */
 import React from 'react';
+import calculate from '../logic/calculate';
 import './calculator.css';
+// import calculate from '../logic/calculate';
 
-// eslint-disable-next-line react/prefer-stateless-function
 class Calculator extends React.Component {
+  constructor(props) {
+    super(props);
+    this.operationHandler = this.operationHandler.bind(this);
+    this.state = {
+      total: 0,
+      next: 0,
+      operation: null,
+    };
+  }
+
+  operationHandler = (e) => {
+    this.setState((state) => calculate(state, e.target.value));
+  }
+
   render() {
+    const { total, next, operation } = this.state;
     return (
       <>
         <div className="row justify-content-center">
           <div className="col-sm-12 col-md-4" id="calc-container">
-            <div className="row justify-content-end px-1 bg-secondary">0</div>
+            <div className="row justify-content-end px-1 bg-secondary">{total} {operation} {next}</div>
             <div className="row justify-content-center align-content-center">
-              <div className="col-3 border">AC</div>
-              <div className="col-3 border">+/-</div>
-              <div className="col-3 border">%</div>
-              <div className="col-3 bg-warning border"><i className="fas fa-divide" /></div>
+              <button onClick={this.operationHandler} className="col-3 border" type="button" value="AC">AC</button>
```

Comment:
- [ ] It seems like there is a bug when a user press the 'AC' button and the display of the calculator disappears. Let's review our logic to fix this issue.